### DISCUSSION TITLE
The "Languages" panel on a repo's home page is useless for evaluating the size of the project

### DISCUSSION THREAD

**Discussion by infogulch** [community/community#112397] [Repositories]
### Select Topic Area

Product Feedback

### Body

Very often when I'm browsing a project I want to quickly understand how much code there is, aka its size. It's very frustrating to have to either blatantly guess (inaccurate, requires mental math) or download and run cloc (wastes time and bandwidth, leaves a mess). I think the best way to do fix this would be to add order-of-magnitude total line count to the Languages panel.

I'm not looking for a high level of precision or accuracy or recency. I don't want an exact number. If I was interested in doing a detailed analysis I'd clone it and run cloc myself. Instead I want to know roughly how big it is, especially if I'm evaluating it in comparison to multiple other repositories I'm unfamiliar with.

Consider the languages panel for these projects:

---

![image](https://github.com/community/community/assets/133882/757f3e03-5787-4800-b291-5815d3cb0251)

![image](https://github.com/community/community/assets/133882/1b9c8aef-5ead-410d-a278-b77485a40f52)

![image](https://github.com/community/community/assets/133882/f72c5075-92cd-4734-8031-0a4044f89b1e)

![image](https://github.com/community/community/assets/133882/0073576f-3dbb-4e81-9068-7f49ec288873)

![image](https://github.com/community/community/assets/133882/df971768-31d8-4855-b541-88684b3b6bb8)

---

This says what languages they're made with, but there's no indication as to how much code there is in total. Now compare to these screenshots that I manually edited to include the total LOC in the available empty space:

![image](https://github.com/community/community/assets/133882/618e42be-bac6-42f3-af70-eaf48d33cb4f)

![image](https://github.com/community/community/assets/133882/de359459-b23b-418a-aec7-8c5dee2bb912)

![image](https://github.com/community/community/assets/133882/dc43e1db-3ed0-4c39-b80c-75430aec28fc)

![image](https://github.com/community/community/assets/133882/c7531d95-95f7-4401-8d69-67188956d471)

![image](https://github.com/community/community/assets/133882/ad81da0a-254e-4a12-a08b-6e4e3bd25d71)

---

These altered examples provide a much richer picture of what a repository contains than percentages alone. Note that with the total, per-file counts are not necessary and I can get an estimate for a given file type by multiplying the total by the percentage if I wanted.

**Comment by spenserblack** [comment]
Good idea! But FYI that the language distributions are calculated by the *sizes* of the files, not LOC, so it will likely have to be "x KB" instead.

**Reply by infogulch** [reply]
I see. That makes sense from a computing standpoint.

Doesn't github index all the files already? Maybe it could count LoC for each file while it's reading through the file anyway, and then come back through and aggregate the counts later?
